src/base/bittorrent/private/sequentialdownloadsguard.cpp

```diff
@@ -0,0 +1,128 @@
+             << "sdo" << (torrentHandle->isSequentialDownloadOverride() ? "[ON]" : "[OFF]");
+
+    if (isSequentialDownloadEnabled(torrentHandle)
+        && (torrentHandle->distributedCopies() > AVAILABILITY_LIMIT)
+        && torrentHandle->isPreviewable())
+        enableSequentialDownload(torrentHandle);
+    else
+        disableSequentialDownload(torrentHandle);
+}
+
+/**
+   @brief Is Sequential download enabled?
+
+   Takes into account global setting in Options dialog and possible override
+   in Add new Torrent dialog.
+ */
+bool SequentialDownloadsGuard::isSequentialDownloadEnabled(TorrentHandle *const torrentHandle) const
+{
+    // If Sequential download was overriden in Add new Torrent dialog, return
+    // that overriden value
+    if (m_session->isSequentialDownload() != torrentHandle->isSequentialDownloadOverride())
+        return torrentHandle->isSequentialDownloadOverride();
+
+    return m_session->isSequentialDownload();
+}
+
+/**
+   @brief Increase QTimer interval after first shot.
+ */
+void SequentialDownloadsGuard::increaseTimerInterval()
+{
+    if (!m_intervalIncreased) {
+        m_sequentialGuardTimer->start(SEQ_DOWNLOAD_REFRESH_INTERVAL);
+        m_intervalIncreased = true;
+    }
+}
+
+/**
+   @brief Enable Sequential download and first/last pieces and log message.
+ */
+void SequentialDownloadsGuard::enableSequentialDownload(TorrentHandle *const torrentHandle)
+{
+    const bool isSequentialDownload = torrentHandle->isSequentialDownload();
+    const bool hasFirstLastPiecePriority = torrentHandle->hasFirstLastPiecePriority();
+
+    // Separated if needed, because of fastresume
+    if (!isSequentialDownload)
+        torrentHandle->setSequentialDownload(true);
+    if (!hasFirstLastPiecePriority)
+        torrentHandle->setFirstLastPiecePriority(true);
+
+    // Log Sequential download Guard's result, only when status changed
+    if (!isSequentialDownload || !hasFirstLastPiecePriority)
+        logResult(torrentHandle);
+}
+
+/**
+   @brief Disable Sequential download and first/last pieces and log message.
+ */
+void SequentialDownloadsGuard::disableSequentialDownload(TorrentHandle *const torrentHandle)
+{
+    if (torrentHandle->isSequentialDownload()) {
+        torrentHandle->setSequentialDownload(false);
+        torrentHandle->setFirstLastPiecePriority(false);
+
+        // Log Sequential download Guard's result, only when status changed
+        logResult(torrentHandle);
+    }
+}
+
+/**
+   @brief Log Sequential download Guard's result.
+ */
+void SequentialDownloadsGuard::logResult(TorrentHandle *const torrentHandle) const
+{
+    QString logMessage = tr("Torrent '%1' : sequential download %2, first/last pieces %3")
+        .arg(torrentHandle->name())
+        .arg(torrentHandle->isSequentialDownload() ? tr("[ON]") : tr("[OFF]"))
+        .arg(torrentHandle->hasFirstLastPiecePriority() ? tr("[ON]") : tr("[OFF]"));
```

Comment:
Maybe 
`QString("[%1]").arg("ON")` to avoid including the `[ ]` in the translation?
But you better wait for more comments on this.

Reply:
@thalieht, @silverqx:  

>avoid including the `[ ]` in the translation?  

I seem to recall that some languages use a different character to replace square-brackets, so this should probably be left as is.

Reply:
>some languages use a different character to replace square-brackets

Oh i didn't know that.